dbt/adapters/snowflake/connections.py

```diff
@@ -443,25 +443,28 @@ def _split_queries(cls, sql):
         split_query = snowflake.connector.util_text.split_statements(sql_buf)
         return [part[0] for part in split_query]
 
-    @classmethod
-    def process_results(cls, column_names, rows):
-        # Override for Snowflake. The datetime objects returned by
-        # snowflake-connector-python are not pickleable, so we need
-        # to replace them with sane timezones
-        fixed = []
+    @staticmethod
+    def _fix_rows(rows: Iterable[Iterable]) -> Iterable[Iterable]:
+        # See note in process_results().
         for row in rows:
             fixed_row = []
             for col in row:
                 if isinstance(col, datetime.datetime) and col.tzinfo:
                     offset = col.utcoffset()
+                    assert offset is not None
                     offset_seconds = offset.total_seconds()
-                    new_timezone = pytz.FixedOffset(offset_seconds // 60)
+                    new_timezone = pytz.FixedOffset(int(offset_seconds // 60))
                     col = col.astimezone(tz=new_timezone)
                 fixed_row.append(col)
 
-            fixed.append(fixed_row)
+            yield fixed_row
 
-        return super().process_results(column_names, fixed)
+    @classmethod
+    def process_results(cls, column_names, rows):
```

Comment:
Do we even need this method anymore, or can we delete it?

Reply:
I did this in Python to check some things for myself:

```
>>> from typing import Iterator

>>> def my_generator():
...     for i in [1, 2, 3, 4, 5]:
...         yield i
...

>>> def get_my_generator():
...     return my_generator()
...

>>> type(my_generator())
<class 'generator'>

>>> type(get_my_generator())
<class 'generator'>

>>> isinstance(my_generator(), Iterable)
True

>>> isinstance(my_generator(), Iterator)
True
```

Conclusions:

The `return` at the end of process_results() is correct, since it is returning a generator (which is an instance of Iterable), not an element of the generator.

Reply:
Interesting, that's not how I would have thought that worked, but it does. TIL